wolfcrypt/src/asn.c

```diff
@@ -36094,7 +36128,7 @@ enum {
 #endif
 
 /* parse crl buffer into decoded state, 0 on success */
-int ParseCRL(DecodedCRL* dcrl, const byte* buff, word32 sz, int verify,
+int ParseCRL(WOLFSSL_CRL* crl, DecodedCRL* dcrl, const byte* buff, word32 sz, int verify,
```

Comment:
Why is this changing to `WOLFSSL_CRL`? You cannot do this. It is a structure in internal.h. A wolfCrypt function should not be reliant on a wolfSSL internal structure.

Also causes re-def errors like crazy: 

```
$ ../testing/git-hooks/wolfssl-multi-test.sh all-gcc-c99
Warning, awk is not Gnu Awk.  Some features may malfunction.
testing with default gcc version 9.4.0
testing with default g++ version 9.4.0
testing without "latest" gcc.
testing without "latest" g++.
testing without clang.
testing with clang-tidy version 12.0.0
testing without cppcheck (version 1.90 is too old).
testing with valgrind-3.15.0
testing with no default linux kernel.
wolfssl-multi-test SHA256 4eabf843d8
uncommitted changes detected but won't be tested.
testing 1 scenario: all-gcc-c99...
using workspace /tmp/wolfssl_test_workdir.818547
refreshing wolfssl tags... done.
refreshing fips... done, now at 04591187ff
refreshing wolfAsyncCrypt... done, now at 99a7a50e76
refreshing wolfsentry... done, now at b8f6c5e607
[skipping refresh of wolf-wireguard-kernel -- repo not found.]
[skipping refresh of wolf-wireguard-tools -- repo not found.]
computing build env hash... done [1cb9dd03d4]
loading results database... done [2346 ents].
testing OID f4bfcccffd [1 of 1]
commit f4bfcccffd20dddca18a72cd3e275a0a4396661a (origin/pr5559, origin/HEAD, pr5559)
Author: Tesfa Mael <tesfa@wolfssl.com>
Date:   Sat Sep 3 16:06:39 2022 -0700

    CRL binary search and document CRL_STATIC_REVOKED_LIST

 src/crl.c                      | 96 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------
 wolfcrypt/src/asn.c            | 56 +++++++++++++++++++++++++++++++++++++++++++++-----------
 wolfssl/crl.h                  |  5 ++++-
 wolfssl/internal.h             | 16 +++++++++++++++-
 wolfssl/wolfcrypt/asn.h        |  2 +-
 wolfssl/wolfcrypt/asn_public.h |  5 ++++-
 6 files changed, 153 insertions(+), 27 deletions(-)
[all-gcc-c99] [1 of 1] [f4bfcccffd]
    autogen.sh f4bfcccffd...   real 0m12.382s  user 0m10.734s  sys 0m0.869s
    configure...   real 0m15.787s  user 0m11.095s  sys 0m6.407s
    build...In file included from examples/client/client.c:35:
./wolfssl/ssl.h:149:37: error: redefinition of typedef ‘WOLFSSL_CRL’ [-Werror=pedantic]
  149 | typedef struct WOLFSSL_CRL          WOLFSSL_CRL;
      |                                     ^~~~~~~~~~~
In file included from ./wolfssl/ssl.h:35,
                 from examples/client/client.c:35:
./wolfssl/wolfcrypt/asn_public.h:84:32: note: previous declaration of ‘WOLFSSL_CRL’ was here
   84 |     typedef struct WOLFSSL_CRL WOLFSSL_CRL;
      |                                ^~~~~~~~~~~
In file included from ./wolfssl/openssl/objects.h:29,
                 from ./wolfssl/openssl/evp.h:1185,
                 from wolfcrypt/test/test.c:319:
./wolfssl/ssl.h:149:37: error: redefinition of typedef ‘WOLFSSL_CRL’ [-Werror=pedantic]
  149 | typedef struct WOLFSSL_CRL          WOLFSSL_CRL;
      |                                     ^~~~~~~~~~~
In file included from ./wolfssl/wolfcrypt/asn.h:67,
                 from wolfcrypt/test/test.c:220:
./wolfssl/wolfcrypt/asn_public.h:84:32: note: previous declaration of ‘WOLFSSL_CRL’ was here
   84 |     typedef struct WOLFSSL_CRL WOLFSSL_CRL;
      |                                ^~~~~~~~~~~
In file included from ./wolfssl/openssl/objects.h:29,
                 from ./wolfssl/openssl/evp.h:1185,
                 from wolfcrypt/src/wc_port.c:69:
./wolfssl/ssl.h:149:37: error: redefinition of typedef ‘WOLFSSL_CRL’ [-Werror=pedantic]
  149 | typedef struct WOLFSSL_CRL          WOLFSSL_CRL;
      |                                     ^~~~~~~~~~~
In file included from ./wolfssl/wolfcrypt/asn.h:67,
                 from ./wolfssl/openssl/ec.h:28,
                 from ./wolfssl/openssl/evp.h:50,
                 from wolfcrypt/src/wc_port.c:69:
./wolfssl/wolfcrypt/asn_public.h:84:32: note: previous declaration of ‘WOLFSSL_CRL’ was here
   84 |     typedef struct WOLFSSL_CRL WOLFSSL_CRL;
      |                                ^~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from wolfcrypt/src/asn.c:182:
./wolfssl/ssl.h:149:37: error: redefinition of typedef ‘WOLFSSL_CRL’ [-Werror=pedantic]
  149 | typedef struct WOLFSSL_CRL          WOLFSSL_CRL;
      |                                     ^~~~~~~~~~~
In file included from ./wolfssl/wolfcrypt/asn.h:67,
                 from wolfcrypt/src/asn.c:93:
./wolfssl/wolfcrypt/asn_public.h:84:32: note: previous declaration of ‘WOLFSSL_CRL’ was here
   84 |     typedef struct WOLFSSL_CRL WOLFSSL_CRL;
      |                                ^~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from src/wolfio.c:37:
./wolfssl/ssl.h:149:37: error: redefinition of typedef ‘WOLFSSL_CRL’ [-Werror=pedantic]
  149 | typedef struct WOLFSSL_CRL          WOLFSSL_CRL;
      |                                     ^~~~~~~~~~~
In file included from ./wolfssl/ssl.h:35,
                 from ./wolfssl/internal.h:28,
                 from src/wolfio.c:37:
./wolfssl/wolfcrypt/asn_public.h:84:32: note: previous declaration of ‘WOLFSSL_CRL’ was here
   84 |     typedef struct WOLFSSL_CRL WOLFSSL_CRL;
      |                                ^~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from src/internal.c:93:
./wolfssl/ssl.h:149:37: error: redefinition of typedef ‘WOLFSSL_CRL’ [-Werror=pedantic]
  149 | typedef struct WOLFSSL_CRL          WOLFSSL_CRL;
      |                                     ^~~~~~~~~~~
In file included from ./wolfssl/ssl.h:35,
                 from ./wolfssl/internal.h:28,
                 from src/internal.c:93:
./wolfssl/wolfcrypt/asn_public.h:84:32: note: previous declaration of ‘WOLFSSL_CRL’ was here
   84 |     typedef struct WOLFSSL_CRL WOLFSSL_CRL;
      |                                ^~~~~~~~~~~
In file included from wolfcrypt/src/asn.c:182:
./wolfssl/internal.h:2146:36: error: redefinition of typedef ‘WOLFSSL_BY_DIR_HASH’ [-Werror=pedantic]
 2146 | typedef struct WOLFSSL_BY_DIR_HASH WOLFSSL_BY_DIR_HASH;
      |                                    ^~~~~~~~~~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from wolfcrypt/src/asn.c:182:
./wolfssl/ssl.h:154:37: note: previous declaration of ‘WOLFSSL_BY_DIR_HASH’ was here
  154 | typedef struct WOLFSSL_BY_DIR_HASH  WOLFSSL_BY_DIR_HASH;
      |                                     ^~~~~~~~~~~~~~~~~~~
In file included from wolfcrypt/src/asn.c:182:
./wolfssl/internal.h:2147:37: error: redefinition of typedef ‘WOLFSSL_BY_DIR_entry’ [-Werror=pedantic]
 2147 | typedef struct WOLFSSL_BY_DIR_entry WOLFSSL_BY_DIR_entry;
      |                                     ^~~~~~~~~~~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from wolfcrypt/src/asn.c:182:
./wolfssl/ssl.h:155:37: note: previous declaration of ‘WOLFSSL_BY_DIR_entry’ was here
  155 | typedef struct WOLFSSL_BY_DIR_entry WOLFSSL_BY_DIR_entry;
      |                                     ^~~~~~~~~~~~~~~~~~~~
In file included from wolfcrypt/src/asn.c:182:
./wolfssl/internal.h:2148:35: error: redefinition of typedef ‘WOLFSSL_DIST_POINT’ [-Werror=pedantic]
 2148 | typedef struct WOLFSSL_DIST_POINT WOLFSSL_DIST_POINT;
      |                                   ^~~~~~~~~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from wolfcrypt/src/asn.c:182:
./wolfssl/ssl.h:224:35: note: previous declaration of ‘WOLFSSL_DIST_POINT’ was here
  224 | typedef struct WOLFSSL_DIST_POINT WOLFSSL_DIST_POINT;
      |                                   ^~~~~~~~~~~~~~~~~~
In file included from src/wolfio.c:37:
./wolfssl/internal.h:2146:36: error: redefinition of typedef ‘WOLFSSL_BY_DIR_HASH’ [-Werror=pedantic]
 2146 | typedef struct WOLFSSL_BY_DIR_HASH WOLFSSL_BY_DIR_HASH;
      |                                    ^~~~~~~~~~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from src/wolfio.c:37:
./wolfssl/ssl.h:154:37: note: previous declaration of ‘WOLFSSL_BY_DIR_HASH’ was here
  154 | typedef struct WOLFSSL_BY_DIR_HASH  WOLFSSL_BY_DIR_HASH;
      |                                     ^~~~~~~~~~~~~~~~~~~
In file included from src/wolfio.c:37:
./wolfssl/internal.h:2147:37: error: redefinition of typedef ‘WOLFSSL_BY_DIR_entry’ [-Werror=pedantic]
 2147 | typedef struct WOLFSSL_BY_DIR_entry WOLFSSL_BY_DIR_entry;
      |                                     ^~~~~~~~~~~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from src/wolfio.c:37:
./wolfssl/ssl.h:155:37: note: previous declaration of ‘WOLFSSL_BY_DIR_entry’ was here
  155 | typedef struct WOLFSSL_BY_DIR_entry WOLFSSL_BY_DIR_entry;
      |                                     ^~~~~~~~~~~~~~~~~~~~
In file included from src/wolfio.c:37:
./wolfssl/internal.h:2148:35: error: redefinition of typedef ‘WOLFSSL_DIST_POINT’ [-Werror=pedantic]
 2148 | typedef struct WOLFSSL_DIST_POINT WOLFSSL_DIST_POINT;
      |                                   ^~~~~~~~~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from src/wolfio.c:37:
./wolfssl/ssl.h:224:35: note: previous declaration of ‘WOLFSSL_DIST_POINT’ was here
  224 | typedef struct WOLFSSL_DIST_POINT WOLFSSL_DIST_POINT;
      |                                   ^~~~~~~~~~~~~~~~~~
In file included from src/internal.c:93:
./wolfssl/internal.h:2146:36: error: redefinition of typedef ‘WOLFSSL_BY_DIR_HASH’ [-Werror=pedantic]
 2146 | typedef struct WOLFSSL_BY_DIR_HASH WOLFSSL_BY_DIR_HASH;
      |                                    ^~~~~~~~~~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from src/internal.c:93:
./wolfssl/ssl.h:154:37: note: previous declaration of ‘WOLFSSL_BY_DIR_HASH’ was here
  154 | typedef struct WOLFSSL_BY_DIR_HASH  WOLFSSL_BY_DIR_HASH;
      |                                     ^~~~~~~~~~~~~~~~~~~
In file included from src/internal.c:93:
./wolfssl/internal.h:2147:37: error: redefinition of typedef ‘WOLFSSL_BY_DIR_entry’ [-Werror=pedantic]
 2147 | typedef struct WOLFSSL_BY_DIR_entry WOLFSSL_BY_DIR_entry;
      |                                     ^~~~~~~~~~~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from src/internal.c:93:
./wolfssl/ssl.h:155:37: note: previous declaration of ‘WOLFSSL_BY_DIR_entry’ was here
  155 | typedef struct WOLFSSL_BY_DIR_entry WOLFSSL_BY_DIR_entry;
      |                                     ^~~~~~~~~~~~~~~~~~~~
In file included from src/internal.c:93:
./wolfssl/internal.h:2148:35: error: redefinition of typedef ‘WOLFSSL_DIST_POINT’ [-Werror=pedantic]
 2148 | typedef struct WOLFSSL_DIST_POINT WOLFSSL_DIST_POINT;
      |                                   ^~~~~~~~~~~~~~~~~~
In file included from ./wolfssl/internal.h:28,
                 from src/internal.c:93:
./wolfssl/ssl.h:224:35: note: previous declaration of ‘WOLFSSL_DIST_POINT’ was here
  224 | typedef struct WOLFSSL_DIST_POINT WOLFSSL_DIST_POINT;
      |                                   ^~~~~~~~~~~~~~~~~~
In file included from examples/benchmark/tls_bench.c:43:
./wolfssl/ssl.h:149:37: error: redefinition of typedef ‘WOLFSSL_CRL’ [-Werror=pedantic]
  149 | typedef struct WOLFSSL_CRL          WOLFSSL_CRL;
      |                                     ^~~~~~~~~~~
In file included from ./wolfssl/ssl.h:35,
                 from examples/benchmark/tls_bench.c:43:
./wolfssl/wolfcrypt/asn_public.h:84:32: note: previous declaration of ‘WOLFSSL_CRL’ was here
   84 |     typedef struct WOLFSSL_CRL WOLFSSL_CRL;
      |                                ^~~~~~~~~~~
In file included from examples/client/client.c:35:
./wolfssl/ssl.h:149:37: error: redefinition of typedef ‘WOLFSSL_CRL’ [-Werror=pedantic]
  149 | typedef struct WOLFSSL_CRL          WOLFSSL_CRL;
      |                                     ^~~~~~~~~~~
In file included from ./wolfssl/ssl.h:35,
                 from examples/client/client.c:35:
./wolfssl/wolfcrypt/asn_public.h:84:32: note: previous declaration of ‘WOLFSSL_CRL’ was here
   84 |     typedef struct WOLFSSL_CRL WOLFSSL_CRL;
      |                                ^~~~~~~~~~~
cc1: all warnings being treated as errors
In file included from ./wolfssl/openssl/objects.h:29,
                 from ./wolfssl/openssl/evp.h:1185,
                 from wolfcrypt/test/test.c:319:
./wolfssl/ssl.h:149:37: error: redefinition of typedef ‘WOLFSSL_CRL’ [-Werror=pedantic]
  149 | typedef struct WOLFSSL_CRL          WOLFSSL_CRL;
      |                                     ^~~~~~~~~~~
In file included from ./wolfssl/wolfcrypt/asn.h:67,
                 from wolfcrypt/test/test.c:220:
./wolfssl/wolfcrypt/asn_public.h:84:32: note: previous declaration of ‘WOLFSSL_CRL’ was here
   84 |     typedef struct WOLFSSL_CRL WOLFSSL_CRL;
      |                                ^~~~~~~~~~~
```